edb/server/protocol/auth_ext/oauth.py

```diff
@@ -39,26 +39,36 @@ def __init__(
         )
 
         provider_config = self._get_provider_config(provider_name)
-        provider_args = (provider_config.client_id, provider_config.secret)
+        provider_args: tuple[str, str] | tuple[str, str, str, str] = (
+            provider_config.client_id,
+            provider_config.secret,
+        )
```

Comment:
maybe we can just inline the following `self.provider = github.GitHubProvider(*provider_args, ...)` in each case branch? I feel like it'll be easier that way

Reply:
yeah, I think you're right that it's both more straightforward, and probably easier to understand. Will fix.